api/clients/disperser_client.go

```diff
@@ -107,6 +107,28 @@ func (c *disperserClient) DisperseBlob(ctx context.Context, data []byte, quorums
 }
 
 func (c *disperserClient) DisperseBlobAuthenticated(ctx context.Context, data []byte, quorums []uint8) (*disperser.BlobStatus, []byte, error) {
+	// first check if signer is valid
+	accountId, err := c.signer.GetAccountID()
```

Comment:
check if `c.signer == nil`?

Reply:
after some reflections, we could have just assign the signer interface as nil if privateKeyHex is not provided. This make NoopSigner impl looks like a over-do. But I kind of like noopSigner because it complies to the interface, and leave less space for error.

Reply:
my main thing about it is because the signer is a interface, if it is a struct pointer, I will fall immediately after nil check

Reply:
Yeah shouldn't it fail immediately if `c.signer` is nil?

Reply:
I did it, and the commit actually check it.

https://github.com/Layr-Labs/eigenda/pull/770/files#diff-80719721baf29a1f6b39d999668efd8ae1938a4d2ece27e1f9889edb486afb87R110

Not sure why github won't show it.